cms/core/db_router.py

```diff
@@ -25,8 +27,21 @@ def __init__(self) -> None:
         if READ_REPLICA_DB_ALIAS not in settings.DATABASES:  # pragma: no cover
             raise ImproperlyConfigured("Read replica is not configured.")
 
+        # Collect targets of GenericRelation to work around Django bug
+        # @see db_for_read
+        self.generic_target_models: set[type[Model]] = set()
+        for model in apps.get_models():
+            for field in model._meta.get_fields():
+                if isinstance(field, GenericRelation) and field.related_model != "self":
+                    self.generic_target_models.add(field.related_model)
+
     def db_for_read(self, model: type[Model], **hints: Any) -> Optional[str]:
         """Determine which database should be used for read queries."""
+        # Models which are targetted by a GenericRelation must use the write connection.
```

Comment:
nitpick: 
```suggestion
        # Models which are targeted by a GenericRelation must use the write connection.
```

or even simpler, "Models used in a GenericRelation must use the write connection"

Reply:
Fixed in https://github.com/ONSdigital/dis-wagtail/pull/191/commits/31ffdfd22b4d11a270ce83c97908e8bca5dde2f2